drivers/dma/dma_mcux_edma.c

```diff
@@ -651,10 +651,39 @@ static int dma_mcux_edma_init(const struct device *dev)
 			    irq_enable(DT_INST_IRQ_BY_IDX(n, idx, irq));	\
 	}
 
+#define EDMA_CHANNELS_SHARED_IRQ_MASK(i, n) \
+		(DT_PROP_BY_IDX(DT_DRV_INST(n), channels_shared_irq_mask, i))
+
+#define EDMA_CHANNELS_MASK(n) static int edma_channel_mask_##n[] = { \
+							LISTIFY(DT_PROP_LEN(DT_DRV_INST(n),	 \
+							channels_shared_irq_mask),           \
+							EDMA_CHANNELS_SHARED_IRQ_MASK,		 \
+							(,), n)}
+
+#define GET_EDMA_CHANNEL_SHARED_IRQ_STEP(n) \
+			(DT_INST_PROP(n, dma_channels) / 32)
+
+#define EDMA_CHANNELS_SHARED_REGISTER_IN_IRQ(idx, n, dev) \
+		uint32_t step = GET_EDMA_CHANNEL_SHARED_IRQ_STEP(n); \
+		for (int _i = 0; _i < step; _i++) {	\
```

Comment:
can you make this macro define a function helper definition so that the preprocessor output is more readable when debugging , otherwise all this code will end up on one line with the rest of the init macro

Reply:
Can't use function to replace macro, because instance index n and  edma_channel_mask_##n array need to be used by EDMA_CHANNELS_SHARED_REGISTER_IN_IRQ. This is a common driver,  driver don't know how many edma instance will exist on one NPI.

Reply:
Updated.